### PR TITLE
Fix compiling issue after rebasing to M108.

### PR DESCRIPTION
**Pull request by jianjunz**
This PR only enables OWT build with libwebrtc 108 on Windows and iOS with clang. Not all features are working properly.

### CODE REVIEW

File: api/media_types.cc [hunk 1]
```diff
@@ -17,6 +17,7 @@ namespace cricket {
 const char kMediaTypeVideo[] = "video";
 const char kMediaTypeAudio[] = "audio";
 const char kMediaTypeData[] = "data";
+const char kMediaTypeScreen[] = "screen";
```
Comment (taste1981) [inline api/media_types.cc]: Since we no longer needs for port range allocation. Could we remove them?

File: api/media_types.h [hunk 2]
```diff
@@ -24,6 +24,7 @@ enum MediaType {
   MEDIA_TYPE_AUDIO,
   MEDIA_TYPE_VIDEO,
   MEDIA_TYPE_DATA,
+  MEDIA_TYPE_SCREEN,
```
Comment (taste1981) [inline api/media_types.h]: ditto

File: call/BUILD.gn [hunk 3]
```diff
@@ -523,7 +523,7 @@ if (rtc_include_tests) {
         "../test/time_controller:time_controller",
         "../video",
         "adaptation:resource_adaptation_test_utilities",
-      "//test/scenario:scenario",
+        "../test/scenario:scenario",
         "//testing/gmock",
```
Comment (taste1981) [inline call/BUILD.gn:527]: why testing/gmock & test/gtest does not need this while scenario will?

File: modules/video_coding/packet_buffer.cc [hunk 4]
```diff
@@ -443,23 +442,26 @@ std::vector<std::unique_ptr<PacketBuffer::Packet>> PacketBuffer::FindFrames(
...
-      for (uint16_t i = start_seq_num; i != end_seq_num; ++i) {
-        std::unique_ptr<Packet>& packet = buffer_[i % buffer_.size()];
-        RTC_DCHECK(packet);
-        RTC_DCHECK_EQ(i, packet->seq_num);
-        // Ensure frame boundary flags are properly set.
-        packet->video_header.is_first_packet_in_frame = (i == start_seq_num);
-        packet->video_header.is_last_packet_in_frame = (i == seq_num);
-        found_frames.push_back(std::move(packet));
-      }
+      if (is_h264 || full_frame_found) {
```
Comment (taste1981) [inline modules/video_coding/packet_buffer.cc:445]: any specific reason h264is excluded from this logic?

File: pc/peer_connection_factory.cc [hunk 5]
```diff
@@ -128,20 +128,14 @@ RtpCapabilities PeerConnectionFactory::GetRtpSenderCapabilities(
           GetDefaultEnabledRtpHeaderExtensions(media_engine()->voice());
       return ToRtpCapabilities(cricket_codecs, extensions);
     }
-    case cricket::MEDIA_TYPE_VIDEO: {
+    case cricket::MEDIA_TYPE_VIDEO:
+    case cricket::MEDIA_TYPE_SCREEN: {
```
Comment (taste1981) [inline pc/peer_connection_factory.cc]: Please remove the screen type?

File: call/BUILD.gn (hunk 3 above)
Comment (jianjunz) [reply call/BUILD.gn:527]: `//testing/gmock` is in root directory, while `test/gtest` is a relative path to `third_party/webrtc`.

File: modules/video_coding/packet_buffer.cc (hunk 4 above)
Comment (jianjunz) [reply modules/video_coding/packet_buffer.cc:445]: No. It just follows the old logical.

File: api/media_types.cc (hunk 1 above)
Comment (jianjunz) [reply api/media_types.cc]: Removed.

File: api/media_types.h (hunk 2 above)
Comment (jianjunz) [reply api/media_types.h]: Removed.

File: pc/peer_connection_factory.cc (hunk 5 above)
Comment (jianjunz) [reply pc/peer_connection_factory.cc]: Removed.

**Review by taste1981 — APPROVED** [review]

**Pull request merged**